### PR TITLE
feat: Graph api, tests

### PR DESCRIPTION
**Pull request by NamelsKing**

### CODE REVIEW

File: src/index.spec.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    // eslint-disable-next-line
+    const initializeFunc = jest.spyOn(BadgerSDK.prototype as any, 'initialize');
+    initializeFunc.mockImplementation(() => new Promise((res) => res(true)));
+
+    sdk = new BadgerSDK({
+      network: Network.Ethereum,
+      provider: ethAlchemyNodeUrl,
+    });
+
+    isSdkReady = Boolean(await sdk.ready());
```
Comment (axejintao) [inline src/index.spec.ts:21]: might need to think more about this test in particular. we should probably mock the individual service methods ready functions as they call call out to external services for success and if we use this cloudflare public rpc its much for flaky and subject to failures - also we are just relying on a third party for test success

**Review by axejintao — COMMENTED** [review]
code lgtm, few comments on tests

File: src/index.spec.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import BadgerSDK, { BadgerAPI, BadgerGraph } from '.';
+import { Network } from './config';
+
+const ethAlchemyNodeUrl =
+  'https://eth-mainnet.alchemyapi.io/v2/-8IubgVIbg-UaXks6VLvgsuYujAravb4';
```
[the remark this answers is not in the document]
Comment (NamelsKing) [reply src/index.spec.ts:5]: ohhh, sry - sure

File: src/index.spec.ts (hunk 1 above)
Comment (NamelsKing) [reply src/index.spec.ts:21]: ok, no point in this test then

Comment (NamelsKing) [reply src/index.spec.ts:21]: i`ll just mock the whole init

**Pull request merged**